std/experimental/checkedint.d

```diff
@@ -473,20 +473,35 @@ if (isIntegral!T || is(T == Checked!(U, H), U, H))
     }
 
     // toHash
-    /** Generates a hash for `this`. If `Hook` defines `hookToHash`, the call
-      immediately returns `hook.hookToHash(get)`. Otherwise, if payload type
-      defines `toHash`, call is redirected to payload's `toHash`, else the
-      built-in function `getHash` is called.
-    */
+    /**
 
-    size_t toHash()
+    Generates a hash for `this`. If `Hook` defines `hookToHash`, the call
+    immediately returns `hook.hookToHash(payload)`, else the `payload`'s type
+    is checked for a `toHash` function. Otherwise, the build-in function
+    `getHash` is called.
+
+    */
+    size_t toHash() const nothrow @safe
     {
         static if (hasMember!(Hook, "hookToHash"))
-            return hook.hookToHash(payload);
+        {
+            static if (stateSize!Hook > 0)
```

Comment:
This isn't consistent with the docs above
> If `Hook` defines `hookToHash`, the call immediately returns `hook.hookToHash(payload)`
The docs don't mention anything about the state of the hook and how this matters in the hash.

This should just forward to the hook, regardless of state or not.
An example for when this would break is the following:
If your hook is a reference counted object, then two references to the same `hook` will
have different hashes because of the reference count value.

Just forward to `hook.hookToHash` and let the hook do all the work; for the previous example,
only the hook can decide to ignore the reference count.